apps/src/codebridge/FileBrowser/index.tsx

```diff
@@ -282,28 +284,79 @@ export const FileBrowser = React.memo(() => {
     setFileType,
   } = useCodebridgeContext();
   const isReadOnly = useAppSelector(isReadOnlyWorkspace);
+  const dialogControl = useDialogControl();
 
-  const newFolderPrompt: FilesComponentProps['newFolderPrompt'] = useMemo(
```

Comment:
this style of refactoring in this component is just breaking apart into a separate function that displays the prompt, and then the existing function which now takes the prompted value as an argument and other proceeds as normal.

But the gist of the change is that it's moving from `window.prompt` -> `dialogControl.showDialog({type : DialogType.GenericPrompt, ... })`